backend/src/main/java/com/woowacourse/friendogly/footprint/service/FootprintCommandService.java

```diff
@@ -77,4 +79,15 @@ private void validateRecentFootprintExists(Long memberId) {
             throw new FriendoglyException(String.format("마지막 발자국을 찍은 뒤 %d초가 경과되지 않았습니다.", FOOTPRINT_COOLDOWN_SECOND));
         }
     }
+
+    public UpdateWalkStatusResponse updateWalkStatus(Long memberId, UpdateWalkStatusRequest request) {
+        Footprint footprint = footprintRepository.findTopOneByMemberIdOrderByCreatedAtDesc(memberId)
+                .orElseThrow(() -> new FriendoglyException("발자국이 존재하지 않습니다."));
+        if (footprint.isDeleted()) {
+            throw new FriendoglyException("가장 최근 발자국이 삭제된 상태입니다.");
+        }
+
+        footprint.changeWalkStatus(new Location(request.latitude(), request.longitude()));
+        return new UpdateWalkStatusResponse(footprint.getWalkStatus());
```

Comment:
```suggestion
        Footprint footprint = footprintRepository.getRecentByMemberId(memberId);
        footprint.changeWalkStatus(new Location(request.latitude(), request.longitude()));
        return new UpdateWalkStatusResponse(footprint.getWalkStatus());
```

```java
// repository
    @Query("""
            SELECT footprint
            FROM Footprint footprint
            WHERE footprint.member.id = :memberId AND footprint.isDeleted = false
            """)
    Optional<Footprint> findRecentByMemberId(@Param("memberId") Long memberId);

    default Footprint getRecentByMemberId(Long memberId) {
        return findRecentByMemberId(memberId)
                .orElseThrow(() -> new FriendoglyException("exception"));
    }
```

1. 다른 도메인에서 하는 것처럼 default 메서드 두고 getBy~ 쓰면 좋겠네요~! 예외 처리 로직 분리 + 가독성까지 챙겨갈 수 있겠어요
2. 논리 삭제 판단은 DB에서 해도 될 것 같아요!

Reply:
@takoyakimchi 정성스러운 코멘트..감동
코드도 다 짜준..
1. getBy한다고 find를 못쓰는 것이 아닌데 혼동하다가 안써버렸네요! 수정하겠습니다
2. 논리 삭제 판단을 service에서 한 이유는 상황에 따라 어떤 service 메서드에서는 삭제된 발자국을 불러와야 할 수도 있다고 생각했기 때문입니다. 
논리 삭제 판단 DB에서 할 경우 -> repository에 논리 삭제된enitity조회용 메서드,  논리 삭제안되것 포함 entity 조회용메서드 이렇게 2개씩 생김
삭제된 발자국을 부를 일이 없다면 repo단으로 내려도 괜찮겠네요. 지금 다른 발자국 기능도 hotfix해야하는데 그 메서드들 hotfix하고도 삭제된 발자국 필요없다면 repo로 내려보겠습니다~!

Reply:
좋아용~~~